### ISSUE TITLE
Help with pattern recognition

### ISSUE THREAD

**Issue by AmycoFr** [microsoft/WindowsVisionSkillsPreview#24]
Hello,
I want to study the use of AI in the context of pattern recognition. Not that of human faces that has been done for a long time, but in that of the recognition of mushroom spores. Visually we can compare these spores to seeds of various plants.
I start in this area but I think it is already building models of these objects.
Can Windows Vision Skills help with this project?

Thank you

**Comment by dhung-msft** [comment]
If you're wanting to bundle your spore recognition model as a Windows Vision Skill, then yes absolutely. Check out our [Sentiment Analyzer sample](https://github.com/microsoft/WindowsVisionSkillsPreview/tree/master/samples/SentimentAnalyzerCustomSkill) to see how to produce and consume a custom skill using an [ONNX ](https://onnx.ai/) machine learning model. You can get ONNX models from the [ONNX model zoo](https://github.com/onnx/models) or by converting to the ONNX format from other ML frameworks (e.g. PyTorch, TensorFlow, etc).

If you're looking for a pre-trained model for recognizing mushroom spores, unfortunately we don't currently have such a model/skill. However, we welcome your contribution!

**Comment by LPBourret** [comment]
> Hello,
> I want to study the use of AI in the context of pattern recognition. Not that of human faces that has been done for a long time, but in that of the recognition of mushroom spores. Visually we can compare these spores to seeds of various plants.
> I start in this area but I think it is already building models of these objects.
> Can Windows Vision Skills help with this project?
> 
> Thank you

Also on top of David's suggestions, check out [Azure Custom Vision Services](https://azure.microsoft.com/en-us/services/cognitive-services/custom-vision-service/) where you can upload images of your spores and train a custom classifier model from them. After you've reached the accuracy you are satisfied with, you can export this model to the ONNX fomat and run it with [WindowsML](https://docs.microsoft.com/en-us/windows/ai/windows-ml/) inside of your application.

-Louis

**Comment by AmycoFr** [comment]
Good track that these Azure Custom Vision Services. I will follow her and try to build models of mushroom spores. In this field a great precision is necessary in the detection of forms.

 

Thank you Louis

Alain

 

De : Louis-Philippe Bourret <notifications@github.com> 
Envoyé : mardi 7 mai 2019 21:21
À : microsoft/WindowsVisionSkillsPreview <WindowsVisionSkillsPreview@noreply.github.com>
Cc : AmycoFr <Amyco@orange.fr>; Author <author@noreply.github.com>
Objet : Re: [microsoft/WindowsVisionSkillsPreview] Help with pattern recognition (#24)

 

Hello,
I want to study the use of AI in the context of pattern recognition. Not that of human faces that has been done for a long time, but in that of the recognition of mushroom spores. Visually we can compare these spores to seeds of various plants.
I start in this area but I think it is already building models of these objects.
Can Windows Vision Skills help with this project?

Thank you

Also on top of David's suggestions, check out Azure Custom Vision Services <https://azure.microsoft.com/en-us/services/cognitive-services/custom-vision-service/>  where you can upload images of your spores and train a custom classifier model from them. After you've reached the accuracy you are satisfied with, you can export this model to the ONNX fomat and run it with WindowsML <https://docs.microsoft.com/en-us/windows/ai/windows-ml/>  inside of your application.

-Louis

**Comment by AmycoFr** [comment]
Thank You dhung-msft for your interesting suggestions.

Regards

**Comment by LPBourret** [comment]
feel free to re-open or open a new issue if you need more help.

-Louis
